repository_service_tuf/cli/admin/import_targets.py

```diff
@@ -0,0 +1,168 @@
+import json
+import os
+from datetime import datetime
+from typing import Any, Dict, List
+
+from rich.console import Console
+from sqlalchemy import Connection, MetaData, Table, create_engine
+from sqlalchemy.exc import IntegrityError
+
+from repository_service_tuf.cli import click
+from repository_service_tuf.cli.admin import admin
+from repository_service_tuf.helpers.api_client import (
+    URL,
+    Methods,
+    get_headers,
+    is_logged,
+    request_server,
+    task_status,
+)
+from repository_service_tuf.helpers.tuf import Metadata, SuccinctRoles
+
+console = Console()
+
+
+def _check_csv_files(csv_files: List[str]):
+    not_found_csv_files: List[str] = []
+    for csv_file in csv_files:
+        if not os.path.isfile(csv_file):
+            not_found_csv_files.append(csv_file)
+
+    if len(not_found_csv_files) > 0:
+        raise click.ClickException(
+            f"CSV file(s) not found: {(', ').join(not_found_csv_files)}"
+        )
+
+
+def _parse_csv_data(
+    csv_file: str, succinct_roles: SuccinctRoles
+) -> List[Dict[str, Any]]:
+    rstuf_db_data: List[Dict[str, Any]] = []
+    with open(csv_file, "r") as f:
+        for line in f:
+            rstuf_db_data.append(
+                {
+                    "path": line.split(";")[0],
+                    "info": {
+                        "length": int(line.split(";")[1]),
+                        "hashes": {line.split(";")[2]: line.split(";")[3]},
+                    },
+                    "rolename": succinct_roles.get_role_for_target(
+                        line.split(";")[0]
+                    ),
+                    "published": False,
+                    "action": "ADD",
+                    "last_update": datetime.now(),
+                }
+            )
+
+    return rstuf_db_data
+
+
+def _import_csv_to_rstuf(
+    db_client: Connection,
+    rstuf_table: Table,
+    csv_files: List[str],
+    succinct_roles: SuccinctRoles,
+) -> None:
+    for csv_file in csv_files:
+        console.print(f"Import status: Loading data from {csv_file}")
+        rstuf_db_data = _parse_csv_data(csv_file, succinct_roles)
+        console.print(f"Import status: Importing {csv_file} data")
+        try:
+            db_client.execute(rstuf_table.insert(), rstuf_db_data)
+        except IntegrityError:
+            raise click.ClickException(
+                "Import status: ABORTED due duplicated targets. "
```

Comment:
We are sure this error will be thrown only when there are duplicating targets.
We have parsed the csv file already, so all format errors should be caught already, but any possible problem?
I am asking because the error message is really specific and for `IntegrityError` in the `sqlalchemy` documentation is [written](https://docs.sqlalchemy.org/en/14/errors.html#integrityerror):
```
IntegrityError
Exception raised when the relational integrity of the database is affected, e.g. a foreign key check fails.

This error is a [DBAPI Error](https://docs.sqlalchemy.org/en/14/errors.html#error-dbapi) and originates from the database driver (DBAPI), not SQLAlchemy itself.
```

Reply:
We validate only if the CSV file exists, and we parse it. We don't look for duplicates, and we don't want do it.

The reason is we can use the insert for that. Once it is about (`click.ClickException`), the insert is not committed to the DB. We fail without changing the DB.
If we add the check before, we need to implement the algorithm to check for duplicates, and the `insert()` will do it again once the DB uses it as *unique*